compose.yaml

```diff
@@ -139,6 +139,8 @@ services:
       AZURE_ABFS_OAUTH_SECRET: ""
       AZURE_ABFS_OAUTH_ENDPOINT: ""
       AZURE_WASB_ACCESS_KEY: ""
+    ports:
+      - 9083:9083
```

Comment:
Why was this necessary? If you're looking to expose the hive metastore to the flink container, do that by adding this container to the flink network, not by exposing the port to the host.

Reply:
Thanks. I actually first tried adding `networks: - flink` for `hive-metastore` but that led to
```
E   Caused by: java.net.ConnectException: Connection refused (Connection refused)
```

Digging deeper into this, I found that `TestConfForStreaming` (the `TestConf` used by the tests in `ibis/backends/flink/tests/`) creates a local Flink environment, which means the tests do not run on the dockerized environment. This is why we have to expose the meta-store to the local env.

I have put the env creation code in a new function `get_table_env(remote_env: bool, streaming_mode: bool)` to make it more transparent and configurable. Switched to remote env in `TestConfForStreaming` to see if we can run the tests without exposing the meta-store port, but that gave me 
```
Caused by: java.util.concurrent.ExecutionException: org.apache.flink.client.program.ProgramInvocationException: Job failed (JobID: 8065e44d3abe8da591f9c169cde351cd)
...
Caused by: java.io.IOException: Failed to create the parent directory: /private/var/folders/9g/7ncp4wjj22q25spxdy7t44080000gn/T/pytest-of-mehmet/pytest-520/test_table0
```
The error seems to be due to the Flink client being on a file system that is not shared with the Flink `TaskManager`. I guess this is why the env was left to be local in `TestConfForStreaming` at the time.